src/ReadLine.Demo/Program.cs

```diff
@@ -15,10 +15,19 @@ public static void Main(string[] args)
 
             ReadLine.AutoCompletionHandler = (t, s) =>
             {
-                if (t.StartsWith("git"))
-                    return new string[] { "init", "clone", "pull", "push" };
-                else
-                    return null;
+                
+                string _command = t.Substring(0, t.Length);
+                switch(_command)
+                {
+                    case "git":
+                        return new string[] { "git init", "git clone", "git pull", "git push", "git add" };
+                    case "ls":
+                        return new string[] { "ls -a", "ls -al"};
+                    case "dotnet":
+                        return new string[] { "dotnet restore", "dotnet run", "dotnet new" };
+                    default:
+                        return null;
+                }
```

Comment:
yeah, I tested it. With this in case, when a user types `git` and the tab key, ReadLine provides possible subcommands, without necessarily substituting the command for the sub subcommand. Like so 
`git` 
would  suggest
```
git add
git pull
git push
```
as opposed to having to type in the `git` twice, which was the case before 
. I think this is all based on preference and would depend on the actual use case.